django/db/models/query.py

```diff
@@ -301,12 +301,15 @@ def __or__(self, other):
     # METHODS THAT DO DATABASE QUERIES #
     ####################################
 
-    def iterator(self):
+    def iterator(self, chunked_fetch=True):
         """
         An iterator over the results from applying this QuerySet to the
         database.
         """
-        return iter(self._iterable_class(self, chunked_fetch=True))
+        use_chunked_fetch = (
+            chunked_fetch and connections[self.db].settings_dict.get('ENABLE_SERVER_SIDE_CURSORS', True)
+        )
+        return iter(self._iterable_class(self, chunked_fetch=use_chunked_fetch))
```

Comment:
That was my first implementation based on Anssi's recommendation. But I find it dangerous to have the `chunked_fetch` argument prevail. I think one would expect `ENABLE_SERVER_SIDE_CURSORS=False` to disable server-side cursors for the connection, not solely if `chunked_fetch in (None, False)`.